rust/cardano-chain-follower/src/chain_sync_config.rs

```diff
@@ -150,8 +150,23 @@ impl ChainSyncConfig {
         // Start the Mithril Snapshot Follower
         let rx = self.mithril_cfg.run().await?;
 
-        // Start Chain Sync
-        *locked_handle = Some(tokio::spawn(chain_sync(self.clone(), rx)));
+        // Wrap inside a panic catcher to detect if the task panics.
+        let result = panic::catch_unwind(|| {
+            stats::start_thread(self.chain, stats::thread::name::CHAIN_SYNC, true);
+            // Start Chain Sync
+            tokio::spawn(chain_sync(self.clone(), rx))
```

Comment:
```suggestion
            tokio::spawn( async move {
                 start_thread(...);
                 let result =  chain_sync(self.clone(), rx);
                 stop_thread(...);
                 result
            })
```
Probably doesn't need catch_unwind here.